### PR TITLE
Add uncheckedCast(x) and currentThreadId() methods to make it easier …

### PR DESCRIPTION
**Pull request by peter-lawrey**
…to support removing warnings in Java 21

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java [hunk 1]
```diff
@@ -32,14 +32,14 @@
 public interface ReferenceCountedTracer extends ReferenceCounted {
 
     /**
-     * Factory method that creates a new instance of {@link ReferenceCountedTracer}. The instance returned
+     * Factory method that creates a new instance of . The instance returned
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java]: This is missing a link

File: src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java [hunk 2]
```diff
@@ -32,14 +32,14 @@
...
      * is based on the resource tracing configuration of the JVM.
      *
      * @param onRelease The {@link Runnable} that will be executed when the object is released.
      * @param uniqueId  A {@link Supplier} of unique identifiers for the object. It should provide a unique
      *                  string identifier for each invocation.
      * @param type      The {@link Class} representing the type of the object being reference counted.
-     * @return A new instance of {@link ReferenceCountedTracer}.
+     * @return A new instance of .
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java]: This is missing a link

File: src/main/java/net/openhft/chronicle/core/io/ReferenceOwner.java [hunk 3]
```diff
@@ -48,14 +48,14 @@ public interface ReferenceOwner {
     ReferenceOwner TMP = new VanillaReferenceOwner("tmp");
 
     /**
-     * Creates and returns a temporary {@link ReferenceOwner} with the given name.
+     * Creates and returns a temporary  with the given name.
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/core/io/ReferenceOwner.java]: Missing a link to a type

File: src/main/java/net/openhft/chronicle/core/io/ReferenceOwner.java [hunk 4]
```diff
@@ -48,14 +48,14 @@ public interface ReferenceOwner {
...
      * <p>
      * When resource tracing is enabled, a new {@link VanillaReferenceOwner} is created with the specified name.
      * Otherwise, the predefined {@link ReferenceOwner#TMP} instance is returned, regardless of the provided name.
      * 
      *
      * @param name The name to be assigned to the temporary reference owner, used for identification and debugging purposes.
-     * @return A temporary {@link ReferenceOwner} instance.
+     * @return A temporary  instance.
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/core/io/ReferenceOwner.java]: Missing a link to a type

**Review by tgd — CHANGES_REQUESTED** [review]

**Review by yevgenp — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java (hunk 1 above)
Comment (yevgenp) [reply src/main/java/net/openhft/chronicle/core/io/ReferenceCountedTracer.java]: It's not. I guess it's inherent reference to the enclosing type/class.
The space is extra though :)

**Review by tgd — APPROVED** [review]

**Pull request merged**